Use GET to validate api resource existence

Rainrider opened the pull request:
This PR does the following:

- adds a new existence verifier using GET requests to verify api resource existence in contrast to the current custom POST endpoint and deprecates the current existence checker
- adds a new idiomatic `ApiResourceExists` validation constraint and deprecates `ApiResourceExist`

Closes #92

Rainrider commented:
@mtarld ~~The tests fail because of the self-deprecation (commented out in the last commit). I don't know how to proceed about that. The deprecation addresses [your comment](https://github.com/mtarld/api-platform-ms-bundle/issues/92#issuecomment-2804089174).~~ Ah, nvm, had to add a legacy group annotation. Still not sure if this is the right way to trigger the deprecation though.

Rainrider commented:
@mtarld any thoughts on this?

mtarld commented on src/ApiResource/ExistenceChecker.php:
```diff
@@ -24,43 +16,25 @@ class ExistenceChecker implements ReplaceableHttpClientInterface
-        $checkedIris = $this->serializer->deserialize($response->getContent(), ApiResourceExistenceCheckerView::class, 'json');
-
-        return $checkedIris->existences;
+        return match ($response->getStatusCode()) {
+            200, 203 => true,
+            404 => false,
+            // make it throw
+            default => (bool) $response->getContent(true),
+        };
```
Let's return true if `>= 200 && < 300`. I'm not sure why we need to throw an exception.

Rainrider replied:
The exception is to support the current behaviour of `ApiResourceExist` (see `skipOnError`)

mtarld commented on src/ApiResource/ExistenceChecker.php:
```diff
@@ -24,43 +16,25 @@ class ExistenceChecker implements ReplaceableHttpClientInterface
 {
```
This file contains BC breaks (removing constructor arguments, renaming a method).
Maybe we should create a new service and deprecate that one?

Rainrider replied:
I just considered the service to be internal, because it is not exposed by its FQCN. I'll implement the new one named `ExistenceVerifier` and deprecate this one then.

I see you would like a smoother upgrade experience, so I could actually keep the old system intact (apart from deprecating it) and just add the new validator and its supporting services. We could stay on 1.x then.

mtarld commented on src/Validator/ApiResourceExistValidator.php:
```diff
@@ -15,13 +15,11 @@ class_exists(\LogicException::class);
  * @author Mathias Arlaud <mathias.arlaud@gmail.com>
  */
-class ApiResourceExistValidator extends ConstraintValidator
+final class ApiResourceExistValidator extends ConstraintValidator
```
I'd create hard final in another PR

mtarld commented on src/Validator/ApiResourceExist.php:
```diff
@@ -5,59 +5,48 @@
-
-        $this->message = $message ?? $this->message;
-        $this->skipOnError = $skipOnError ?? $this->skipOnError;
+        trigger_deprecation('mtarld/api-platform-ms-bundle', '2.0.0', sprintf('Use %s instead', ApiResourceExists::class));
```
`2.0.0` should instead be the latest 1.x version

mtarld replied:
```suggestion
        trigger_deprecation('mtarld/api-platform-ms-bundle', '2.0.0', sprintf('%s is deprecated, use %s instead', ApiResourceExist::class, ApiResourceExists::class));
```

mtarld reviewed (changes requested):
First round :wink:

Rainrider committed:
feat: add ApiResourceExists validation constraint

Replaces ApiResourceExist

Rainrider commented:
@mtarld thank your for the feedback, I hope I addressed everything. I updated the PR description.

mtarld commented on tests/Validator/ApiResourceExistsValidatorTest.php:
```diff
@@ -0,0 +1,127 @@
+    public static function provideBlankIris(): iterable
+    {
+        yield 'null' => [null];
+        yield 'empty' => [''];
```
An empty string as IRI should fail to me.

Rainrider replied:
Hm. virtually all default constraint validators fall through on `null` and emptry string (except [PasswordStrengthValidator](https://github.com/symfony/symfony/blob/195f1f123d1328ef534f70e4605948049120ae73/src/Symfony/Component/Validator/Constraints/PasswordStrengthValidator.php#L35-L37)?). This was the reason I included `''`. I'll remove it then.

mtarld replied:
Yes you're right, let's keep it like that

mtarld reviewed (commented):
This is way better like that! Just one little comment.
Thanks @Rainrider 🙂

mtarld commented:
Btw, do you have an account on SymfonyDevs slack?

Rainrider commented:
It appears I don't. https://symfony.com/slack appears to be inactive, so I don't know how to join.

Rainrider commented:
Ah, static analysis fails now, but this is not introduced by me. I could fix this in a separate PR together with some other house keeping if you don't mind.

mtarld commented:
> It appears I don't. [symfony.com/slack](https://symfony.com/slack) appears to be inactive, so I don't know how to join.

Can you try with https://symfony-devs.slack.com/? Then reach me at `mtarld`

> Ah, static analysis fails now, but this is not introduced by me. I could fix this in a separate PR together with some other house keeping if you don't mind.

With pleasure!

Rainrider commented:
@mtarld https://symfony-devs.slack.com there is no sign-up under this link, only sign-in. I cannot create an account.

mtarld commented:
Alright, then, can you reach me at my email? (Can be found in https://github.com/mtarld/api-platform-ms-bundle#authors)

Rainrider commented:
I rebased on 1.x and added an override attribute to `ApiResourceExistsvalidator::validate`. This should pass the tests once #95 is merged. Alternatively you could change the target branch of this PR to `house-keeping` to test this claim.
